### ISSUE TITLE
Percent array literal inside a block causes infinite loop with `--disable-uncorrectable`.

### ISSUE THREAD

**Issue by sliiser** [rubocop/rubocop#12208]
A rather simple example that causes an infinite loop with several cops. The example is given with `Metrics/MethodLength`, but is also reproducible with `Metrics/ClassLength`, `Metrics/AbcSize`, `Metrics/CyclomaticComplexity` and `Metrics/PerceivedComplexity`. List is not exhaustive.

--------
## Steps to reproduce the problem

```yml
# .rubocop.yml
AllCops:
  NewCops: enable

Metrics/MethodLength:
  Max: 2 # just for demonstration
```
```ruby
# frozen_string_literal: true

def foo
  bar do
    %w[]
  end
end
```
`$ rubocop -A --disable-uncorrectable`

## Expected behavior

```ruby
# frozen_string_literal: true

def foo # rubocop:todo Metrics/MethodLength
  bar do
    %w[]
  end
end
```

## Actual behavior

```ruby
# frozen_string_literal: true

def foo
  bar do
    # rubocop:todo Metrics/MethodLength
    # rubocop:todo Metrics/MethodLength
    # rubocop:todo Metrics/MethodLength
    ...
    # rubocop:todo Metrics/MethodLength
    # rubocop:todo Metrics/MethodLength
    # rubocop:todo Metrics/MethodLength
    %w[]
    # rubocop:enable Metrics/MethodLength
    # rubocop:enable Metrics/MethodLength
    # rubocop:enable Metrics/MethodLength
    ...
    # rubocop:enable Metrics/MethodLength
    # rubocop:enable Metrics/MethodLength
    # rubocop:enable Metrics/MethodLength
  end
end

```
`Infinite loop detected in <the file>.rb and caused by Metrics/MethodLength`

## RuboCop version

```bash
$ rubocop -V
1.56.3 (using Parser 3.2.2.3, rubocop-ast 1.28.1, running on ruby 3.2.2) [arm64-darwin22]
$ ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]
```
